### ISSUE TITLE
`network-mode: container:<container_name_without_prefix>` does not work

### ISSUE THREAD

**Issue by pyaillet** [srl-labs/containerlab#1731]
Hi!
Thank you for making containerlab, it's a great project to get into networking.

## Problem

I found a problem when trying to use `network-mode: container:<container_name_without_prefix>`.

The early validation of the container name [here](https://github.com/srl-labs/containerlab/blob/main/runtime/docker/docker.go#L929) works.
But later when trying to get the container status [there](https://github.com/srl-labs/containerlab/blob/main/runtime/docker/docker.go#L973) the cID used is the raw one.

I added some warn locally to make sure I identified the problem correctly and got this:
```
[...]
WARN[0002] Container [worker2] not found
INFO[0002] node "server2" depends on external container "worker2", which is not running yet. Waited 2s. Retrying...
[...]
```
You can see that the container is not found at this step, and that it's the short name which is used, not the prefixed one.

Even though the early identification of the container worked and the prefixed container is still running:
```
$ docker container ls
[...]
dcabb4c3fc8f   kindest/node:v1.27.3   "/usr/local/bin/entr…"   About an hour ago   Up About an hour                               clab-bgp-cplane-demo-worker2
```

I could try to send a PR to fix this, but I'm not sure which way you would like to handle this.
I can think of two methods:
1. Isolate the [code used to find the container name](https://github.com/srl-labs/containerlab/blob/main/runtime/docker/docker.go#L929-L947) in a dedicated method and reuse it
2. Change the container name in the configuration to reflect the real one

The first solution seems easier but does not guarantee that the problem won't happen again elsewhere, I am not sure to know all the impacts of the second one.

## Workaround

As a simple workaround, it's still possible to use the full container name.

**Comment by hellt** [comment]
Hi @pyaillet 
if the container you're attaching to is not defined in the clab file, then you must use container name as seen in `docker ps` output (aka full container name).

Is your _parent_ container defined in the clab file or is it launched outside of clab?

**Comment by steiler** [comment]
Please provide the topology file.

**Comment by pyaillet** [comment]
@hellt the container is not defined in the clab file. It's created with [kind](https://kind.sigs.k8s.io/), but it used to work with previous versions like 0.31.1 with the attached files.

@steiler I attached the topology file and although the cluster configuration that I use with kind

topo.yaml
```yaml
name: minimal-reproduction
topology:
  kinds:
    linux:
      cmd: bash
  nodes:
    tor:
      kind: linux
      image: frrouting/frr:v8.2.2
      exec:
        # peer over this link with cilium
        - ip addr add 172.0.0.1/24 dev net0
        # Boiler plate to make FRR work
        - touch /etc/frr/vtysh.conf
        - sed -i -e 's/bgpd=no/bgpd=yes/g' /etc/frr/daemons
        - /usr/lib/frr/frrinit.sh start
        - >-
          vtysh -c 'conf t'
          -c 'debug bgp neighbor-events'
          -c 'debug bgp updates'
          -c 'debug bgp zebra'
          -c '! peer with Cilium'
          -c 'router bgp 65000'
          -c ' no bgp ebgp-requires-policy'
          -c ' bgp router-id 172.0.0.1'
          -c ' neighbor 172.0.0.2 remote-as 65001'
          -c ' address-family ipv6 unicast'
          -c '  neighbor 172.0.0.2  activate'
          -c ' exit-address-family'
          -c '!'
    cp:
      kind: linux
      image: nicolaka/netshoot:latest
      network-mode: container:control-plane
      exec:
        # peer over this link with tor
      - ip addr add 172.0.0.2/24 dev net0
  links:
    - endpoints: ["tor:net0", "cp:net0"]
```

cluster.yaml
```yaml
kind: Cluster
name: clab-minimal-reproduction
apiVersion: kind.x-k8s.io/v1alpha4
networking:
  disableDefaultCNI: true
nodes:
- role: control-plane
  kubeadmConfigPatches:
  - |
    kind: InitConfiguration
    nodeRegistration:
      kubeletExtraArgs:
        node-ip: "172.0.0.2"
containerdConfigPatches:
- |-
  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:5000"]
    endpoint = ["http://kind-registry:5000"]
```

However if you think that it was just by chance that it worked in older version, and that it's not really a feature, I can still use the full container name.

**Comment by hellt** [comment]
Let us have a closer look @pyaillet 
Maybe indeed we can make short name to work (again), but using a full name is an easy solution to get you unblocked.

**Comment by steiler** [comment]
@pyaillet Took a look at the whole scenario now.

**TLDR;** The implementation is correct as it is right now.

So what you can do is reference same topology file based containers via its short name like so:
cp references l1's network namespace. And this works perfectly fine.
```
name: ext-test
topology:
  nodes:
    l1:
      kind: linux
      image: alpine:latest
      cmd: sleep infinity
    cp:
      kind: linux
      image: alpine:latest
      network-mode: container:l1
      cmd: sleep infinity

  links:
    - endpoints: ["l1:eth1", "cp:eth2"]
```
What you described here ist that you made use of the nameing schema and from the `docker ps` command perspective the kind cluster looks like it was created via the topology. (Although it is missing clab specific labels etc.).

External containers even though referenced in the network-mode. Must be referenced via its full name. If that stuff did work at some point in time it was pure luck and not intended to work that way. Sorry.
